src/AutoMerge/pull-requests.jl

```diff
@@ -95,51 +99,40 @@ function pull_request_build(api::GitHub.GitHubAPI,
     end
 
     if is_new_package(pr)
-        registry_master = clone_repo(registry)
-        if !master_branch_is_default_branch
-            checkout_branch(registry_master, master_branch)
-        end
-        pull_request_build(api,
-                           NewPackage(),
-                           pr,
-                           current_pr_head_commit_sha,
-                           registry;
-                           auth = auth,
-                           authorized_authors=authorized_authors,
-                           authorized_authors_special_jll_exceptions=authorized_authors_special_jll_exceptions,
-                           error_exit_if_automerge_not_applicable = error_exit_if_automerge_not_applicable,
-                           registry_head = registry_head,
-                           registry_master = registry_master,
-                           suggest_onepointzero = suggest_onepointzero,
-                           whoami=whoami,
-                           registry_deps = registry_deps)
-        rm(registry_master; force = true, recursive = true)
+        registration_type = NewPackage()
     elseif is_new_version(pr)
-        registry_master = clone_repo(registry)
-        if !master_branch_is_default_branch
-            checkout_branch(registry_master, master_branch)
-        end
-        pull_request_build(api,
-                           NewVersion(),
-                           pr,
-                           current_pr_head_commit_sha,
-                           registry;
-                           auth = auth,
-                           authorized_authors=authorized_authors,
-                           authorized_authors_special_jll_exceptions=authorized_authors_special_jll_exceptions,
-                           error_exit_if_automerge_not_applicable = error_exit_if_automerge_not_applicable,
-                           registry_head = registry_head,
-                           registry_master = registry_master,
-                           suggest_onepointzero = suggest_onepointzero,
-                           whoami=whoami,
-                           registry_deps = registry_deps)
-        rm(registry_master; force = true, recursive = true)
+        registration_type = NewVersion()
     else
         throw_not_automerge_applicable(
             AutoMergeNeitherNewPackageNorNewVersion,
             "Neither a new package nor a new version. Exiting...";
             error_exit_if_automerge_not_applicable = error_exit_if_automerge_not_applicable,
         )
+        return nothing
+    end
+
+    registry_master = clone_repo(registry)
+    if !master_branch_is_default_branch
+        checkout_branch(registry_master, master_branch)
     end
+    pkg, version = parse_pull_request_title(registration_type, pr)
+    data = GitHubAutoMergeData(;api,
```

Comment:
I'd like to keep compatibility with Julia 1.3. The syntax `foo(; bar, baz)` isn't available in Julia 1.3, so unfortunately you'll need to write it out as `foo(; bar = bar, baz = baz)`.

Reply:
I know (the PR bumps the compat) but if we're not ready to abandon 1.3 I'll change it. There's a lot of code in this package that would gain from that particular 1.5 feature.

Reply:
Hmmmmmmmmmm.

Reply:
Let's drop 1.3 and 1.4.

I know that people are running AutoMerge on private registries, but I think it's reasonable for them to upgrade AutoMerge to Julia 1.5 or 1.6 if they want new AutoMerge features.

Can you also edit the GitHub Actions workflow files so that the unit tests are run on:
- `'1.5'`
- `'1'`
- `'nightly'`

Reply:
Well, I changed it to 1.3 compatible. Let's leave it like that for now, then we can introduce that feature all over the package at once after this PR is finished.

Reply:
Sounds good.